### PR TITLE
Product Subscriptions: Enable Downloadable setting

### LINKED ISSUE #11218: Downloadable files
Allow adding downloadable files to subscription products.

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Closes: #11218 

## Description

Allows enabling the "Downloadable files" option in the Product form from Product settings. 

## Testing instructions

**Prerequisite**
1. Install Woo Subscription plugin on your store
1. Create a subscriptions product from `wp-admin`

**Steps**
1. Enable the feature flag subscriptionProducts and build the app.
2. On the app, log in to a store with Woo Subscription plugin set up.
3. Navigate to Products tab > select "+" > select manual option > Subscription product.
4. Tap on `...` > `Product settings`
5. You should see toggle "Downloadable Product"
6. Enable it, and you should see the "Downloadable files" option to upload files in the product form
7. Try uploading files and save the product

## Screenshots

Simulator Screen Recording - iPhone 14 Pro - 2023-11-21 at 11 13 11

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by hafizrahman — COMMENTED** [review]
Works correctly in my test 👍🏼 

I was wondering why variation subscription products don't have this setting (both iOS and Android). After doing some digging, it turns out our apps don't support downloadable in any variation product anyway so we're skipping it for subscriptions as well (ref: p1700057058858509/1700057044.683779-slack-C03L1NF1EA3), so all good here.

File: WooCommerce/WooCommerceTests/ViewRelated/Products/Actions Factory/ProductFormActionsFactoryTests.swift [hunk 1]
```diff
@@ -741,6 +741,59 @@ final class ProductFormActionsFactoryTests: XCTestCase {
         XCTAssertEqual(factory.bottomSheetActions(), expectedBottomSheetActions)
     }
 
+    func test_view_model_for_subscription_product_when_downloadable() {
```
Comment (hafizrahman) [inline WooCommerce/WooCommerceTests/ViewRelated/Products/Actions Factory/ProductFormActionsFactoryTests.swift]: np: maybe `test_view_model_for_subscription_product_when_is_downloadable()` flows better as a sentence?

same with the next one: `test_view_model_for_subscription_product_when_is_not_downloadable()`

Comment (selanthiraiyan) [reply WooCommerce/WooCommerceTests/ViewRelated/Products/Actions Factory/ProductFormActionsFactoryTests.swift]: Done in https://github.com/woocommerce/woocommerce-ios/pull/11231/commits/f15934cdb408510add5802429dc4056c3e35170a

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductSettingsSections.swift [hunk 2]
```diff
@@ -21,19 +21,14 @@ enum ProductSettingsSections {
...
-                        ProductSettingsRows.Visibility(settings),
-                        ProductSettingsRows.CatalogVisibility(settings)]
-            }
+            let shouldShowVirtualProductSetting = settings.productType == .simple
+            let shouldShowDownloadableProductSetting = settings.productType == .simple || settings.productType == .subscription
+            let rows: [ProductSettingsRowMediator?] = [ProductSettingsRows.Status(settings),
+                    ProductSettingsRows.Visibility(settings),
+                    ProductSettingsRows.CatalogVisibility(settings),
+                    shouldShowVirtualProductSetting ? ProductSettingsRows.VirtualProduct(settings) : nil,
+                                                      shouldShowDownloadableProductSetting ? ProductSettingsRows.DownloadableProduct(settings) : nil]
```
Comment (hafizrahman) [inline WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductSettingsSections.swift]: just a np but this line seems to have too many whitespace?

I like the logic simplification here, though!

**Review by hafizrahman — APPROVED** [review]
It's all good! I sent some comments but feel free to merge after having a look 👍🏼

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductSettingsSections.swift (hunk 2 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductSettingsSections.swift]: Fixed indentation in 8d9407b72467709d312a747592159086e7fd732d

**Pull request merged**